Update conversion error notifications to show error details

EricE147 opened the pull request:
# Description

Updated `addConversion`,  `submitEditedConversion`, and `deleteConversion` catch statements in conversions.ts to show error details.

Partly Addresses #869.

1. Add: ![image](https://user-images.githubusercontent.com/113533298/227423717-1d0786f4-4ab1-49c9-a297-7882834faf88.png)
2. Edit: ![image](https://user-images.githubusercontent.com/113533298/227423751-fa7358cd-3999-4601-ac17-4946523de22d.png)
3. Delete: ![image](https://user-images.githubusercontent.com/113533298/227423773-c6ccb87c-fcea-42fb-926f-6b716506236a.png)

All tests currently pass: 
![OED_Tests](https://user-images.githubusercontent.com/113533298/227425372-7e711c9a-fa9e-422f-872d-6fb41036a70d.png)



## Type of change

This PR should not require any major update.

- [ ] Note merging this changes the node modules
- [ ] Note merging this changes the database configuration.
- [ ] This change requires a documentation update

## Checklist


- [x] I have followed the [OED pull request](https://openenergydashboard.github.io/developer/pr.html) ideas
- [x] I have removed text in ( ) from the issue request

## Limitations

1. The time components of `showSuccessNotification` and `showErrorNotification` don't seem to actually change how long the messages stay up as the webpage immediately reloads regardless of how long `autoDismiss` is set for. Another issue may need to be created to fix this. 
2. We were also unable to trigger the error notification naturally and instead forced it using a throw statement so we don't know exactly what the error messages look like.

EricE147 committed:
Update conversion error notifications to show error details

Co-authored-by: Juan Contreras <xJU4Nx@users.noreply.github.com>
Co-authored-by: Jaime Palacios <ArtP1@users.noreply.github.com>

xJU4Nx reviewed (approved):
Waiting for review. Thanks.

salhernandez commented:
Hi @huss 👋🏼 

I mentored the students that address #869. They found a bug with `showSuccessNotification` and `showErrorNotification` where the `autoDismiss` argument seems to not considered. It didn't matter if we set `autoDismiss` to 2 seconds or 2000 seconds, There may be a fixed autoDismiss value deeper into the call stack. Should the students create a issue report for it?

huss commented on src/client/app/actions/conversions.ts:
```diff
@@ -99,7 +99,7 @@ export function submitEditedConversion(editedConversion: t.ConversionData, shoul
 			} catch (err) {
 				// Failure! ):
-				showErrorNotification(translate('conversion.failed.to.edit.conversion'));
+				showErrorNotification(translate('conversion.failed.to.edit.conversion') + '"' + err + '"');
```
In other places in OED we do "err.response.data" to see the actual message. I think it may be a little risky if another type of error but I think we should do it here and in the other messages to be consistent.

huss commented on src/client/app/actions/conversions.ts:
```diff
@@ -150,7 +150,7 @@ export function deleteConversion(conversion: t.ConversionData): Thunk {
 				showSuccessNotification(translate('conversion.successfully.delete.conversion'));
 			} catch (err) {
-				showErrorNotification(translate('conversion.failed.to.delete.conversion'));
+				showErrorNotification(translate('conversion.failed.to.delete.conversion') + '"' + err + '"');
```
This was not your doing but this message does not show up as expected because src/server/routes/conversions.js near the end of the file for the delete route does not use failure but res.status. This happens in two places. Would you be willing to fix that?

If you do edit that file then you could also fix the addConversion route that has
validationResult.error.toString()
but it should be
validationResult.errors.toString()
as done elsewhere. I missed that when the code was added to OED in the past. It causes an issue so the message is not seen.

huss replied:
I fixed to use success/failure and you did the rest so this is fixed.

huss reviewed (changes requested):
Thanks to @EricE147, @ArtP1 & @xJU4Nx for their first PR to OED. Overall, this is good work. I made a couple of comments you can address (or not for the one outside the code you touched) but they are minor.

Just so you know, I tested your code by setting the sourceID to -1 in the route for conversions (src/server/routes/conversions.js):
req.body.sourceId = -1;
This causes the validation to fail and the failure to happen. It can be done in all three of the routes (add, edit, delete) to test each one.
Just for the record, note that I first tried to set the value in src/client/app/actions/conversions.ts but that caused errors in the component since it uses this id to get the string displayed for the source id in 2 of the 3 cases so that is not a good way to do it.

huss commented:
> I mentored the students that address #869. They found a bug with `showSuccessNotification` and `showErrorNotification` where the `autoDismiss` argument seems to not considered. It didn't matter if we set `autoDismiss` to 2 seconds or 2000 seconds, There may be a fixed autoDismiss value deeper into the call stack. Should the students create a issue report for it?

First, thank you @salhernandez for taking the time to mentor these students. As for the noted this issue, it impacts several parts of the code in OED. Thus, I think putting in an issue is a good idea and I would be pleased to have the students do that. I have been thinking about possible solutions and have an idea. Once the issue is posted, I will comment on why it exists and a first cut at a solution.

EricE147 committed:
Update error messages to show response data

EricE147 committed:
Fix typo

EricE147 commented:
@huss we updated the error message to include `response.data` and also fixed the typo in `validationResult.error.toString()`. We tried changing the `res.status` to `failure` but we were still not getting the correct behavior so we left it unchanged.

EricE147 commented:
This is what is shown with `err.response.data`:
![image](https://user-images.githubusercontent.com/113533298/228409499-2fb3eecb-8018-42db-b0d2-9e9acabcde4d.png)

huss committed:
update messages

- Use single quotes as usual in OED.
- Use back quote for multiple string items.
- Use success/failure functions for conversion deletion.
- Make messages and variable name consistent across cases.
- Remove unneeded toString().

huss commented:
That you for your updates. They all looked great. I'm sorry it took a few days as other thing took up my time.

While looking at the code I noticed inconsistencies in previous code (not what you edited) so I made some changes that are in the commit message. This PR is now ready to merge. Congratulations to  @EricE147, @ArtP1 & @xJU4Nx for their first merged code.
